lib/opal/compiler.rb

```diff
@@ -48,17 +49,17 @@ class Compiler
 
     # All compare method nodes - used to optimize performance of
     # math comparisons
-    COMPARE = %w[< > <= >=]
+    COMPARE = %w[< > <= >=].freeze
```

Comment:
Why rubocop add or remove `freeze` ? What is the rule ?

Reply:
Ruby < 2.x (the one that introduced magic `freeze` comment, not sure which exactly):
All others constants must be frozen.

Ruby > 2.x:
Strings don't need to be frozen if there's a `frozen_string_literal: true` comment

Reply:
I think the frozen trick is just for strings, but probably Rubocop is trying to enforce the fact that a _constant_ shouldn't change, and `.freeze` helps guarding against that